sources/packages/api/src/database/entities/designation-agreement.model.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  OneToMany,
+  PrimaryGeneratedColumn,
+  RelationId,
+} from "typeorm";
+import {
+  DesignationAgreementLocation,
+  DesignationAgreementStatus,
+  Institution,
+  User,
+} from ".";
+import { ColumnNames, TableNames } from "../constant";
+import { dateOnlyTransformer } from "../transformers/date-only.transformer";
+import { RecordDataModel } from "./record.model";
+
+/**
+ * Designation agreement created by the institution for the Ministry assessment.
+ * The Ministry will check the data provide and approve of deny the request,
+ * providing also start/end dates and removing/adding locations.
+ */
+@Entity({ name: TableNames.DesignationAgreements })
+export class DesignationAgreement extends RecordDataModel {
+  @PrimaryGeneratedColumn()
+  id: number;
+  /**
+   * Institution id that submitted the designation agreement.
+   */
+  @RelationId((designation: DesignationAgreement) => designation.institution)
+  institutionId: number;
+  /**
+   * Institution that submitted the designation agreement.
+   */
+  @ManyToOne(() => Institution, { eager: false, cascade: false })
+  @JoinColumn({
+    name: "institution_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  institution: Institution;
+  /**
+   * Dynamic data that represents the designation agreement requested by the Institution to be approved by the Ministry.
+   */
+  @Column({
+    name: "submitted_data",
+    type: "jsonb",
+    nullable: false,
+  })
+  submittedData: any;
```

Comment:
Can we define an interface to avoid any ?

Reply:
The data is going to be the result of an area in the form.io that is not required to have any constraints and could be freely changed by the Ministry. The data under this particular section is also not supposed to be used anywhere in the API, only to populate the same form again for visualization, so, I do not see a reason for this particular case.

Reply:
@dheepak-aot, I added more comments to the property because developers in the future can have the same quetion. Hope that helps 😉 
![image](https://user-images.githubusercontent.com/61259237/152608681-8e326a87-0f49-4b42-9d83-752fc9ed9f47.png)